drivers/usb/device/usb_dc_stm32.c

```diff
@@ -69,27 +69,10 @@
 #define EP_TYPE_INTR PCD_EP_TYPE_INTR
 #endif
 
-#ifndef CONFIG_USB_NUM_IN_ENDPOINTS
-#define CONFIG_USB_NUM_IN_ENDPOINTS 0
-#endif /* CONFIG_USB_NUM_IN_ENDPOINTS */
-
-#ifndef CONFIG_USB_NUM_OUT_ENDPOINTS
-#define CONFIG_USB_NUM_OUT_ENDPOINTS 0
-#endif /* CONFIG_USB_NUM_OUT_ENDPOINTS */
-
-/* Total in ep number = bidirectional ep number + in ep number */
-#define NUM_IN_EP (CONFIG_USB_NUM_BIDIR_ENDPOINTS + \
-		   CONFIG_USB_NUM_IN_ENDPOINTS)
-
-/* Total out ep number = bidirectional ep number + out ep number */
-#define NUM_OUT_EP (CONFIG_USB_NUM_BIDIR_ENDPOINTS + \
-		    CONFIG_USB_NUM_OUT_ENDPOINTS)
-/*
- * Total bidirectional ep number = bidirectional ep number + (out ep number +
- * in ep number) / 2.  Because out ep number = in ep number,
- * total bidirectional ep number = total out ep number or total in ep number
- */
-#define NUM_BIDIR_EP NUM_OUT_EP
+/* Total bidirectional endpoints, including endpoint 0 */
+#ifndef CONFIG_USB_NUM_BIDIR_ENDPOINTS
```

Comment:
We don't need this extra check. `num-bidir-endpoints` property is required in DT, so DT parser should handle it and raise an error if it is missing.